src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -109,23 +122,94 @@ private static synchronized void closeResources()  {
       zoneIdToTable.clear();
       zoneIdToTable = null;
     }
-    if (fixedTransitions != null) {
-      fixedTransitions.close();
-      fixedTransitions = null;
+    if (transitions != null) {
+      transitions.close();
+      transitions = null;
     }
   }
 
-  public static ColumnVector fromTimestampToUtcTimestamp(ColumnVector input, ZoneId currentTimeZone) {
-    // TODO: Remove this check when all timezones are supported
-    // (See https://github.com/NVIDIA/spark-rapids/issues/6840)
-    if (!isSupportedTimeZone(currentTimeZone)) {
-      throw new IllegalArgumentException(String.format("Unsupported timezone: %s",
-          currentTimeZone.toString()));
+  private static long getScalefactor(ColumnVector input){
+    DType inputType = input.getType();
+    if (inputType == DType.TIMESTAMP_SECONDS){
+      return 1;
+    } else if (inputType == DType.TIMESTAMP_MILLISECONDS){
+      return 1000;
+    } else if (inputType == DType.TIMESTAMP_MICROSECONDS){
+      return 1000*1000;
+    }
+    throw new UnsupportedOperationException("Input Type is not of TIMESTAMP type");
+  }
+
+  public static boolean isSupportedTimeZone(String zoneId) {
+    try {
+      getZoneId(zoneId); // check that zoneId is a valid zone
+      return true;
+    } catch (ZoneRulesException e) {
+      return false;
+    }
+  }
+
+  // enforce that all timestamps, regardless of timezone, be less than the desired date
+  private static boolean isValidInput(ColumnVector input, ZoneId zoneId){
```

Comment:
The function name seems misleading a bit. It makes me think that the `input` will be invalid if this function returns false. But it seems to actually indicate that the computation should be done in CPU instead. Perhaps `fallBackToCpu` is a better name?